src/coreclr/src/debug/createdump/crashinfomac.cpp

```diff
@@ -31,6 +31,45 @@ CrashInfo::CleanupAndResumeProcess()
     }
 }
 
+static
+kern_return_t
+SuspendMachThread(thread_act_t thread, int tid)
+{
+    kern_return_t result;
+
+    while (true)
```

Comment:
How can you ensure this will not become a hotspot? Basically you poll the thread for suspension making sure that it's not in a non-interruptible operation. Is there any guarantee the thread will eventually trap? Maybe a poll count limit?

Reply:
I took this code from the PAL's macOS exception code.  I assumed it couldn't get stuck in the loop, but maybe I should have a retry count.